cluster/v1beta1/helpers.go

```diff
@@ -117,18 +119,29 @@ type PlacementDecisionGetter interface {
 
 // +k8s:deepcopy-gen=false
 type PlacementDecisionClustersTracker struct {
-	placement                 *Placement
-	placementDecisionGetter   PlacementDecisionGetter
-	existingScheduledClusters sets.Set[string]
-	lock                      sync.RWMutex
+	placement                      *Placement
+	placementDecisionGetter        PlacementDecisionGetter
+	existingScheduledClusters      sets.Set[string]
+	existingScheduledClusterGroups map[GroupKey]sets.Set[string]
+	clustesGroupsIndexToName       map[int32]string
+	clustesGroupsNameToIndex       map[string][]int32
+	lock                           sync.RWMutex
 }
 
-func NewPlacementDecisionClustersTracker(placement *Placement, pdl PlacementDecisionGetter, existingScheduledClusters sets.Set[string]) *PlacementDecisionClustersTracker {
+// +k8s:deepcopy-gen=false
+type GroupKey struct {
```

Comment:
Would it make sense to track the placement decision name and `resourceVersion` to know when it may have changed in the middle of the rollout or after the fact?

This could be used as a "shortcut" during a controller's initial list query of applicable PlacementDecision objects.

Reply:
I'm not sure if need this, since if the user defines rollout strategy and has placement defined, the controller should watch the placement and decision. When the decision changes, should trigger a reconcile to call GetRolloutCluster().